JuliaStream.jl/src/CUDAStream.jl

```diff
@@ -0,0 +1,154 @@
+include("Stream.jl")
+using CUDA
+
+const CuData = StreamData{T,CuArray{T}} where {T}
+const TBSize = 1024::Int
+const DotBlocks = 256::Int
+
+function devices()::Vector{DeviceWithRepr}
+  return !CUDA.functional(false) ? String[] :
+         map(d -> (d, "$(CUDA.name(d)) ($(repr(d)))", "CUDA.jl"), CUDA.devices())
+end
+
+function make_stream(
+  arraysize::Int,
+  scalar::T,
+  device::DeviceWithRepr,
+  silent::Bool,
+)::Tuple{CuData{T},Nothing} where {T}
+
+  if arraysize % TBSize != 0
+    error("arraysize ($(arraysize)) must be divisible by $(TBSize)!")
+  end
+
+  # so CUDA's device is 0 indexed, so -1 from Julia
+  CUDA.device!(device[1])
+  selected = CUDA.device()
+  # show_reason is set to true here so it dumps CUDA info 
+  # for us regardless of whether it's functional
+  if !CUDA.functional(true)
+    error("Non-functional CUDA configuration")
+  end
+  if !silent
+    println("Using CUDA device: $(CUDA.name(selected)) ($(repr(selected)))")
+    println("Kernel parameters: <<<$(arraysize ÷ TBSize),$(TBSize)>>>")
+  end
+  return (
+    CuData{T}(
+      CuArray{T}(undef, arraysize),
+      CuArray{T}(undef, arraysize),
+      CuArray{T}(undef, arraysize),
+      scalar,
+      arraysize,
+    ),
+    nothing,
+  )
+end
+
+function init_arrays!(data::CuData{T}, _, init::Tuple{T,T,T}) where {T}
+  CUDA.fill!(data.a, init[1])
```

Comment:
You shouldn't need the module prefix here, as `fill!` is just a Base method that dispatches correctly since you're passing a CuArray argument already. Only when that's not the case (e.g. with `fill`, which just takes an integer) you need the CUDA-specific version.